dotnet/samples/Concepts/TextGeneration/HuggingFace_TextGeneration.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.HuggingFace;
+using xRetry;
+
+#pragma warning disable format // Format item can be simplified
+#pragma warning disable CA1861 // Avoid constant arrays as arguments
+
+namespace Examples;
+
+// The following example shows how to use Semantic Kernel with HuggingFace API.
+public class HuggingFace_TextGeneration(ITestOutputHelper helper) : BaseTest(helper)
+{
+    /// <summary>
+    /// This example uses HuggingFace Inference API to access hosted models.
+    /// More information here: <see href="https://huggingface.co/inference-api"/>
+    /// </summary>
+    [Fact]
+    public async Task RunInferenceApiExampleAsync()
+    {
+        WriteLine("\n======== HuggingFace Inference API example ========\n");
+
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddHuggingFaceTextGeneration(
+                model: TestConfiguration.HuggingFace.ModelId,
+                apiKey: TestConfiguration.HuggingFace.ApiKey)
+            .Build();
+
+        var questionAnswerFunction = kernel.CreateFunctionFromPrompt("Question: {{$input}}; Answer:");
+
+        var result = await kernel.InvokeAsync(questionAnswerFunction, new() { ["input"] = "What is New York?" });
+
+        WriteLine(result.GetValue<string>());
+    }
+
+    [RetryFact(typeof(HttpOperationException))]
+    public async Task RunStreamingExampleAsync()
+    {
+        WriteLine("\n======== HuggingFace zephyr-7b-beta streaming example ========\n");
+
+        const string Model = "HuggingFaceH4/zephyr-7b-beta";
```

Comment:
Should we read this from config and provide this as a default suggestion there?

Reply:
Not all hugging face public inference API supports Streaming, I added this as a fixed for ensuring a working example, I can add a config and use this as a default. This is the original code, I can change it.